src/integrations/stripe.ts

```diff
@@ -162,6 +143,86 @@ export async function setupPaymentMethodAndPay(
   return null;
 }
 
+/**
+ * Used to either collect payment directly or launch stripe checkout to collect payment.
+ */
+export async function collectPayment(
+  actionData: ActionData,
+  stripePromise = getStripe()
+) {
+  const {
+    client,
+    triggerElement, // action button that triggered this
+    formattedFields,
+    updateFieldValues,
+    integrationData: stripeConfig
+  } = actionData;
+
+  // sync fields to BE
+  await syncFieldChanges(client, formattedFields, stripeConfig);
+
+  // setup any payment method on this step
+  if (actionData.targetElement) {
+    const payMethodResult = await setupPaymentMethod(
```

Comment:
what if payment method has already been set up? is that handled well?

Reply:
This is specifically called out in the spec and is a manual test case that I will test for.  See: https://docs.google.com/document/d/1ADF3b3u-u008Jyhws6g9DT9VEdYzLkQFLNltiLQAhAs/edit#heading=h.fneum7xxs0q9 under the runtime behavior bullet points.